src/main/java/bio/terra/service/resourcemanagement/BufferService.java

```diff
@@ -106,9 +130,29 @@ public ResourceInfo handoutResource() {
       } else {
         throw new BufferServiceAPIException(e);
       }
+    } catch (GeneralSecurityException e) {
+      throw new GoogleResourceException("Failed to cleanup after failing to re-folder project", e);
     }
   }
 
+  private void refolderProjectToSecureFolder(String projectId)
+      throws IOException, GeneralSecurityException {
+    CloudResourceManager cloudResourceManager = googleResourceManagerService.cloudResourceManager();
+    var project = cloudResourceManager.projects().get(projectId).execute();
+
+    ResourceId resourceId =
+        new ResourceId().setType("folder").setId(googleConfig.getSecureFolderResourceId());
+
+    project.setParent(resourceId);
+
+    cloudResourceManager.projects().update(projectId, project).execute();
```

Comment:
Is there any error handling that we may want to do here?

Reply:
errors are handled by the caller